src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1067,86 +1055,312 @@ def inner_product_matrix(self):
+            []
+            sage: QuaternionAlgebra(L, -1, -1).ramified_primes()
+            [Fractional ideal (2, 1/2*a + 1/2), Fractional ideal (2, 1/2*a - 1/2)]
 
-            sage: QuaternionAlgebra(QQ, -58, -69).ramified_primes()
-            [3, 23, 29]
+            sage: x = polygen(ZZ, 'x')
+            sage: F.<a> = NumberField(x^2 - x - 1)
+            sage: QuaternionAlgebra(F, 2*a, F(-1)).ramified_primes()
+            [Fractional ideal (2)]
+
+            sage: QuaternionAlgebra(RR(2.),1).ramified_primes()
+            Traceback (most recent call last):
+            ...
+            ValueError: base field must be rational numbers or a number field
         """
-        if not is_RationalField(self.base_ring()):
-            raise ValueError("base field must be the rational numbers")
+        return self.ramified_places(inf=False)
+
+    @cached_method
+    def discriminant(self):
+        r"""
+        Return the discriminant of this quaternion algebra, i.e. the
+        product of the finite places it ramifies at.
+
+        OUTPUT:
+
+        The discriminant of this quaternion algebra (which has to be defined
+        over a number field), as an element of `\ZZ` if the quaternion algebra
+        is defined over `\QQ`, and as a fractional ideal in the ring of
+        integers of the base number field otherwise.
+
+        EXAMPLES::
+
+            sage: QuaternionAlgebra(210, -22).discriminant()
+            210
+            sage: QuaternionAlgebra(19).discriminant()
+            19
+            sage: QuaternionAlgebra(-1, -1).discriminant()
+            2
+
+            sage: K = QuadraticField(3)
+            sage: L = QuadraticField(-15)
+            sage: QuaternionAlgebra(K, -1, -1).discriminant()
+            Fractional ideal (1)
+            sage: QuaternionAlgebra(L, -1, -1).discriminant()
+            Fractional ideal (2)
+
+            sage: x = polygen(ZZ, 'x')
+            sage: F.<a> = NumberField(x^2 - x - 1)
+            sage: QuaternionAlgebra(F, 2*a, F(-1)).discriminant()
+            Fractional ideal (2)
+
+            sage: QuaternionAlgebra(QQ[sqrt(2)], 3, 19).discriminant()                  # needs sage.symbolic
+            Fractional ideal (1)
+
+            sage: QuaternionAlgebra(RR(2.),1).discriminant()
+            Traceback (most recent call last):
+            ...
+            ValueError: base field must be rational numbers or a number field
+        """
+        F = self.base_ring()
+        if is_RationalField(F):
+            return ZZ.prod(self.ramified_places(inf=False))
 
-        return sorted([p for p in set([2]).union(prime_divisors(self._a.numerator()),
-                prime_divisors(self._a.denominator()), prime_divisors(self._b.numerator()),
-                prime_divisors(self._b.denominator())) if hilbert_symbol(self._a, self._b, p) == -1])
+        return F.ideal(F.prod(self.ramified_places(inf=False)))
 
     def is_isomorphic(self, A) -> bool:
         """
```

Comment:
```suggestion
        r"""
```

Reply:
Is there a reason to prefer this over putting a double backslash for LaTeX commands? I thought that achieved the same result

Reply:
I think the built documentation shows otherwise? It’s displayed as plain QQ right now.

Reply:
With the documentation built on the newest commit? It's wrong in the documentation preview because that one is still built on an old commit, I believe

Reply:
It seems to display correctly in the new build, so I'll resolve all related suggested changes.